new/authentication_config_file.md

```diff
@@ -0,0 +1,73 @@
+---
+RFC: unassigned
+Title: Authentication Configuration
+Author: Thom May <thom@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Authentication Configuration
+
+Currently, the Chef community maintains many tools to aid in the
+management and switching of Chef Server configuration options - for
+example `knife-block`. Any non-ruby tools must also implement their own
+configuration management, since our configuration files must be parsed
+as though they were ruby.
+It would be delightful to provide a single configuration file that was
+solely responsible for handling authentication options, and was in a
+language agnostic format, such as TOML.
+
+## Motivation
+
+    As an Operations Engineer,
+    I want to manage nodes on many Chef Servers,
+    so that I can interact with my whole estate.
+
+    As a tool developer,
+    I want to use the language of my choice,
+    so that I can effectively build delightful Chef features.
+
+## Specification
+
+A new file, `~/.chef/credentials`, will be supported by the chef client
+libraries, and other implementations. It is a TOML file, containing one
+or more sections, each associated with a profile.
+
+```
+[default]
+node_name = "barney"
+client_key = "barney_rubble.pem"
+chef_server_url = "https://api.chef.io/organisations/bedrock"
+
+[dev]
+node_name = "admin"
+client_key = "admin.pem"
+validator_key = "test-validator.pem"
```

Comment:
Can we list fully which keys will be allowed here? Or say "anything goes but not all Chef options will be expressible in TOML", though even in a latter case we could then document which options will always be expressible in here.

Reply:
I think anything goes is probably _correct_, but there may well be value in just documenting the above to make it easier, and then say that any complex config should go in `config.rb`.